lib/utils.js

```diff
@@ -75,19 +78,60 @@ exports.console = Object.freeze(jpmConsole);
 /**
  * Returns the `package.json` manifest as an object
  * from the `cwd`, or `null` if not found.
+ * If you pass in an optional XPI filename, the manifest will be returned
+ * from this directory after a temporary XPI extraction.
  *
- * @return {Object}
+ * @param {String} optional XPI filename
+ * @return {Promise} resolves to a manifest object
  */
-function getManifest () {
+function getManifest (xpiPath) {
   return when.promise(function(resolve, reject) {
-    var cwd = process.cwd();
-    var json = path.join(cwd, "package.json");
-    var manifest = {};
-    try {
-      manifest = require(json);
+    if (xpiPath) {
+      return resolve(getXPIManifest(xpiPath));
+    } else {
+      var cwd = process.cwd();
+      var json = path.join(cwd, "package.json");
+      var manifest = {};
+      try {
+        manifest = require(json);
+      }
+      catch (e) {}
```

Comment:
Looks like the `catch` can come after the `}`. I see both forms in the codebase but this file has a `} catch(e) {` in it so this should probably be consistent with that (and I'm glad that this file has the other one to support my case :wink:).